polygeist/tools/cgeist/driver.cc

```diff
@@ -901,8 +924,23 @@ static void processInputFiles(const cl::list<std::string> &inputFiles,
   
   // Generate MLIR for the C/C++ files.
   std::string fn = (!syclKernelsOnly) ? cfunction.getValue() : "";
-  parseMLIR(Argv0, files, fn, includeDirs, defines, module, triple, DL,
-            commands);
+  parseMLIR(Argv0, files, fn, includeDirs, defines, module, deviceModule,
+            triple, DL, commands);
+}
+
+static bool hasKernels(mlir::gpu::GPUModuleOp deviceModule) {
+  return !deviceModule.getRegion().getOps<mlir::gpu::GPUFuncOp>().empty();
+}
+
+static std::pair<mlir::OwningOpRef<mlir::ModuleOp>,
+                 mlir::OwningOpRef<mlir::gpu::GPUModuleOp>>
+buildModules(mlir::MLIRContext &context) {
+  constexpr llvm::StringLiteral SYCLModuleName{"sycl"};
+
+  mlir::OpBuilder Builder(&context);
+  const auto loc = Builder.getUnknownLoc();
+  return {mlir::ModuleOp::create(loc),
```

Comment:
So you want to name only the GPUModule with the name "sycl". Not sure that is appropriate because:
   - longer term the GPUModule should also possibly contain OpenMP device code, and
   - the CPU module (ModuleOp) might contain SYCL code as well if a GPU is not available ?

Reply:
I think `device_functions` (or simply `device`) is a better name than SYCL because of the first reason you mentioned.

SYCL can also offload kernels to the CPU (a CPU can also be a SYCL device), so the CPU module should not contain SYCL code (with "SYCL code", I mean code to be run in a SYCL device).

Reply:
I like `device` as a name.